### PR TITLE
[feature](Nereids): add command for updating mv with partitions

### PR DESCRIPTION
**Pull request by keanji-x**
## Proposed changes

add command for updating mv with partitions

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by XieJiann** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/UpdateMvByPartitionCommand.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+
+    /**
+     * Construct command
+     *
+     * @param mv materialize view
+     * @param partitions update partitions in mv and tables
+     * @param tableWithPartKey the partitions key for different table
+     * @return command
+     */
+    public UpdateMvByPartitionCommand from(MTMV mv, Set<PartitionItem> partitions,
```
[the remark this answers is not in the document]
Comment (XieJiann) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/UpdateMvByPartitionCommand.java:68]: yes. I'll fixed it

**Comment by XieJiann** [comment]
run buildall

**Comment by XieJiann** [comment]
run buildall

**Review by zddr — APPROVED** [review]
LGTM

**Review by seawinde — APPROVED** [review]

File: fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/UpdateMvByPartitionCommand.java [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+     * @param mv materialize view
+     * @param partitions update partitions in mv and tables
+     * @param tableWithPartKey the partitions key for different table
+     * @return command
+     */
+    public static UpdateMvByPartitionCommand from(MTMV mv, Set<PartitionItem> partitions,
+            Map<OlapTable, String> tableWithPartKey) {
+        NereidsParser parser = new NereidsParser();
+        Map<OlapTable, Set<Expression>> predicates =
+                constructTableWithPredicates(partitions, tableWithPartKey);
```
Comment (morrySnow) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/trees/plans/commands/UpdateMvByPartitionCommand.java:73]: why not set partition list in relation directly

**Review by morrySnow — APPROVED** [review]

**Pull request merged**